tsc/scripts/git_count.py

```diff
@@ -0,0 +1,495 @@
+            if "message" in response and  "API rate limit exceeded" in response["message"]:
+                # Retry due to API rate limit
+                wait_for(t)
+                dt = datetime.now()
+                debugprint("{} Sleep and retry for  {}".format(dt, cmd))
+            else:
+                lookupid = False
+        except:
+            lookupid = False
+            print("{} : {}".format(cmd, output))
+    return response
+
+#
+# Util function: get company info via github id. If it fails, try its committer email address
+#
+def get_company_from_id(id, email):
+
+    # Use github id to find out company name from profile
+    url = "https://api.github.com/users/{}".format(id)
+    response = lookup_with_retry(url)
+
+    #
+    # NOTFOUND: default failure reason, profile is found, but no company info is set in profile
+    #
+    src = "NOTFOUND_COMPANYINFO"
+    if  "message" in response and  "Not Found" in response["message"]:
+        #
+        # NOTFOUND: Can't find profile for a given id
+        #
+        src = "NOTFOUND_PROFILE"
+        
+    if "company" in response:
+        comp = response["company"]
+        if comp == None:
+            if id not in email or "noreply.github.com" not in email:
+                #
+                # NOTFOUND: User sets company  as None in profile
+                #            
+                ret, comp = use_email_to_find_company(email, "NOTFOUND_COMPANYISNONE")
+            else:
+                comp = "NOTFOUND_COMPANYISNONE"
+
+        comp = fixup_companyname(comp)
+    else:
+        ret, comp = use_email_to_find_company(email, src)
+
+    return comp
+
+#
+# Company name mapping
+#
+def company_name_check(key):
+    key = key.lower()
+    if "azure" in key:
+        return True, "MICROSOFT"    
+    if "microsoft" in key:
+        return True, "MICROSOFT"
+    if "nephosinc" in key:
+        return True, "NEPHOSINC"
+    if "intel" in key:
+        return True, "INTEL"
+    if "edge-core" in key:
+        return True, "EDGE-CORE"
+    if "metaswitch" in key:
+        return True, "METASWITCH"
+    if "google" in key:
+        return True, "GOOGLE"
+    if "alibaba" in key:
+        return True, "ALIBABA"
+    if "mellanox" in key:
+        return True, "NVIDIA"
+    if "arista" in key:
+        return True, "ARISTA"
+    if "dell" in key:
+        return True, "DELL"
+    if "linkedin" in key:
+        return True, "LINKEDIN"
+    if "broadcom" in key:
+        return True, "BROADCOM"    
+    if "accton" in key:
+        return True, "ACCTON"
+    if "thebollingers" in key:
+        return True, "THEBOLLINGERS"
+    if "cisco" in key:
+        return True, "CISCO"
+    if "nvidia" in key:
+        return True, "NVIDIA"
+    if "nokia" in key:
+        return True, "NOKIA"
+    if "barefootnetworks" in key:
+        return True, "INTEL"
+    if "innovium" in key:
+        return True, "INNOVIUM"
+    if "cavium" in key:
+        return True, "CAVIUM"
+    
```

Comment:
```suggestion
    if "cavium" in key:
        return True, "CAVIUM"
    if "xflow" in key:
        return True, "XFLOW"

```

Please include xFlow Research as it's both a General Member and one of the contributing organizations. Please see https://sonic-net.github.io/SONiC/index.html for reference